src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -742,60 +745,30 @@ public int Seek(ref Vector<byte> byte0Vector, ref Vector<byte> byte1Vector, ref
         }
 
         /// <summary>
-        /// Find first byte
+        /// Locate the first of the found bytes
         /// </summary>
         /// <param  name="byteEquals"></param >
         /// <returns>The first index of the result vector</returns>
-        /// <exception cref="InvalidOperationException">byteEquals = 0</exception>
-        internal static int FindFirstEqualByte(ref Vector<byte> byteEquals)
+        // Force inlining (64 IL bytes, 91 bytes asm) Issue: https://github.com/dotnet/coreclr/issues/7386
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static int LocateFirstFoundByte(ref Vector<byte> byteEquals)
         {
-            if (!BitConverter.IsLittleEndian) return FindFirstEqualByteSlow(ref byteEquals);
-
-            // Quasi-tree search
             var vector64 = Vector.AsVectorInt64(byteEquals);
-            for (var i = 0; i < Vector<long>.Count; i++)
+            long longValue = 0;
+            var i = 0;
+            for (; i < Vector<long>.Count; i++)
```

Comment:
Is `break;` and `continue;` really faster than just checking `longValue != 0` here? If it's that way to avoid the initial check, you could use do/while. I prefer loop conditions if possible.

If we're trying to avoid comparisons, `i < Vector<long>.Count` also seems unnecessary since `byteEquals` is guaranteed to have something set based on the calling contract.